ext/standard/head.c

```diff
@@ -201,59 +207,154 @@ PHPAPI int php_setcookie(zend_string *name, zend_string *value, time_t expires,
 	return result;
 }
 
+static int php_head_parse_cookie_options_array(zval *options, zend_long *expires, zend_string **path, zend_string **domain, zend_bool *secure, zend_bool *httponly, zend_string **samesite) {
+	int found = 0;
+	zend_string *key;
+	zval *value;
+
+	ZEND_HASH_FOREACH_STR_KEY_VAL(Z_ARRVAL_P(options), key, value) {
+		if (key) {
+			ZVAL_DEREF(value);
+			if(!strcasecmp("expires", ZSTR_VAL(key))) {
+				*expires = zval_get_long(value);
+				found++;
+			} else if(!strcasecmp("path", ZSTR_VAL(key))) {
+				*path = zval_get_string(value);
+				found++;
+			} else if(!strcasecmp("domain", ZSTR_VAL(key))) {
+				*domain = zval_get_string(value);
+				found++;
+			} else if(!strcasecmp("secure", ZSTR_VAL(key))) {
+				*secure = zval_is_true(value);
+				found++;
+			} else if(!strcasecmp("httponly", ZSTR_VAL(key))) {
+				*httponly = zval_is_true(value);
+				found++;
+			} else if(!strcasecmp("samesite", ZSTR_VAL(key))) {
+				*samesite = zval_get_string(value);
+				found++;
+			} else {
+				php_error_docref(NULL, E_WARNING, "Unrecognized key '%s' found in the options array", ZSTR_VAL(key));
+			}
+		} else {
+			php_error_docref(NULL, E_WARNING, "Numeric key found in the options array");
+		}
+	} ZEND_HASH_FOREACH_END();
+
+	/* Array is not empty but no valid keys were found */
+	if (found == 0 && zend_hash_num_elements(Z_ARRVAL_P(options)) > 0) {
+		php_error_docref(NULL, E_WARNING, "No valid options were found in the given array");
+		return 0;
+	}
```

Comment:
This warning seems a bit redundant. Isn't it only thrown if one of the above warnings already triggered?

Reply:
While it's true that it will only trigger when one of the others also triggered, I think it has some usefulness to distinguish between some keys being invalid and all keys being invalid. OTOH, it does feel a bit redundant given that it will only show paired with one of the above. If we remove the warning, should we also remove the `return 0;`? As it stands, the cookie won't be set when this warning is shown.